src/main.cpp

```diff
@@ -5530,6 +5530,25 @@ bool ProcessMessages(CNode* pfrom)
     return fOk;
 }
 
+class CompareInvMempoolOrder
+{
+    CTxMemPool *mp;
+public:
+    CompareInvMempoolOrder(CTxMemPool *mempool)
+    {
+        mp = mempool;
+    }
+
+    bool operator()(const CInv &a, const CInv &b)
+    {
+        if (a.type != MSG_TX) {
```

Comment:
It probably does change it (the sort isn't stable).  I'll make it sort them by height... or I could split transactions off entirely.